test/spec/modules/komoonaBidAdapter_spec.js

```diff
@@ -1,152 +1,164 @@
-    });
-
-    it('requires paramters to make request', () => {
-      adapter.callBids({});
-      expect(requests).to.be.empty;
-    });
-
-    it('requires placementid and hbid', () => {
-      let backup = pbConfig.bids[0].params;
-      pbConfig.bids[0].params = {placementid: 1234}; // no hbid
-      adapter.callBids(pbConfig);
-      expect(requests).to.be.empty;
+  };
 
-      pbConfig.bids[0].params = {hbid: 1234}; // no placementid
-      adapter.callBids(pbConfig);
-      expect(requests).to.be.empty;
-
-      pbConfig.bids[0].params = backup;
-    });
-
-    it('sends bid request to ENDPOINT via POST', () => {
-      adapter.callBids(pbConfig);
-      expect(requests[0].url).to.equal(ENDPOINT);
-      expect(requests[0].method).to.equal('POST');
-    });
+  it('Verifies komoonaAdapter bidder code', () => {
+    expect(spec.code).to.equal('komoona');
   });
 
-  describe('response handler', () => {
-    let server;
-
-    beforeEach(() => {
-      server = sinon.fakeServer.create();
-      sinon.stub(bidmanager, 'addBidResponse');
-    });
-
-    afterEach(() => {
-      server.restore();
-      bidmanager.addBidResponse.restore();
-    });
-
-    it('registers bids', () => {
-      server.respondWith(JSON.stringify(RESPONSE));
-
-      adapter.callBids(REQUEST);
-      server.respond();
-      sinon.assert.calledOnce(bidmanager.addBidResponse);
-
-      const response = bidmanager.addBidResponse.firstCall.args[1];
-      expect(response).to.have.property('statusMessage', 'Bid available');
-      expect(response).to.have.property('cpm', 0.5);
-    });
-
-    it('handles nobid responses', () => {
-      server.respondWith(JSON.stringify({
-        'bids': [{
-          'cpm': 0,
-          'creative': '',
-          'uuid': '30e5e911c00703'
-        }]
-      }));
-
-      adapter.callBids(REQUEST);
-      server.respond();
-      sinon.assert.calledOnce(bidmanager.addBidResponse);
-
-      const response = bidmanager.addBidResponse.firstCall.args[1];
-      expect(response).to.have.property(
-        'statusMessage',
-        'Bid returned empty or error response'
-      );
-    });
+  it('Verifies komoonaAdapter bid request validation', () => {
+    expect(spec.isBidRequestValid(bidsRequest[0])).to.equal(true);
+    expect(spec.isBidRequestValid(bidsRequest[1])).to.equal(true);
+    expect(spec.isBidRequestValid({})).to.equal(false);
+    expect(spec.isBidRequestValid({ params: {} })).to.equal(false);
+    expect(spec.isBidRequestValid({ params: { hbid: 12345 } })).to.equal(false);
+    expect(spec.isBidRequestValid({ params: { placementid: 12345 } })).to.equal(false);
+    expect(spec.isBidRequestValid({ params: { hbid: 12345, placementId: 67890 } })).to.equal(true);
+    expect(spec.isBidRequestValid({ params: { hbid: 12345, placementId: 67890, floorPrice: 0.8 } })).to.equal(true);
+  });
 
-    it('handles JSON.parse errors', () => {
-      server.respondWith('');
+  describe('Verify komoonaAdapter build request', () => {
```

Comment:
These test cases need to be `it` block for 2 reasons.
1) karma is not showing actual count of tests in logs. So even though this is getting executed but count is not correct. I placed it in `it` block and test count was incremented.
2) If this test fails than it stops here itself.

Reply:
True comment and I accept it - we actually use it for internal products test, not sure why describe was used (probably just copied from old test).

It is fixed now